### ISSUE TITLE
[3.3.5/master] Spell: BasePoint should be double instead int32

### ISSUE THREAD

**Issue by Keader** [TrinityCore/TrinityCore#20620]
**Description:**

Some spells need has double values in BasePoint.
Current scripts cause truncate errors.

E.g.
> Keader: I have a question about Fel Armor + Demonic Aegis. One of effects of Fel Armor is: In addition, you regain 2% of your maximum health every 5 sec. But Demonic Aegis add 30% in all effects of Demonic Armor and Fel Armor...
> Keader: but 30% of 2 = 0.6, and BasePoints is a int32, core will truncate 2.6 back to 2
> Keader: it is correct?
> Keader: because fel armor tooltip change when you apply points in demonic aegis ( In addition, you regain 2% to 3% of your maximum health every 5 sec. )
> Keader: Shauren?
> Shauren: yes, the truncation is a real issue
> Keader: :/ any idea how fix it without change everything to float? xD
> Shauren: you cant.
> Shauren: changing everything to doubles IS the answer (also needed in master)

**Branch(es):** Both

**TC rev. hash/commit:** da553a2e54f33149707d8df5acbae04eb163f756

**Comment by ariel-** [comment]
Shouldn't float be better, performance wise?

Alternate solution would be calculating percent to get real basepoints and then apply bonuses on top of those (altough it would complicate matters a little)

**Comment by Shauren** [comment]
No, on x86 (and x86_64) doubles are faster (floats would only be faster in SSE but thats not the case with spells)

**Comment by ariel-** [comment]
Whatever happened to https://github.com/TrinityCore/TrinityCore/pull/17577#issuecomment-232376659 ? 😈 

> because fuck doubles, thats why
